src/main/java/io/quarkus/gizmo2/impl/GizmoImpl.java

```diff
@@ -98,6 +101,18 @@ public StringConstant stringConstant(String value) {
         return (StringConstant) constants.computeIfAbsent(value, StringConstant::new);
     }
 
+    public ByteConstant byteConstant(Byte value) {
+        return (ByteConstant) funnyHashCodeConstants.computeIfAbsent(new ByteConstant(value), Function.identity());
```

Comment:
Shouldn't we try to find a better name for `funnyHashCodeConstants`? :thinking:

Reply:
I would personally merge `constants` and `funnyHashCodeConstants` together (using the `funnyHashCodeConstants` approach), not sure why they're separate (except perhaps a little less allocations in the `constants` case?).